qcodes/instrument/visa.py

```diff
@@ -258,3 +259,21 @@ def snapshot_base(self, update: bool = True,
         snap['timeout'] = self.timeout.get()
 
         return snap
+
+    def wait_until_complete(self, check_after: float, sleep_time: float) -> None:
+        """
+        A utility function that asynchronously queries the Event Status Register (ESR)
+        and checks if the ``0`` bit (the Operation Complete Query (OPC) bit) is set, i.e.,
+        the operation is complete. This function repeatedly returns the content of the ESR
+        and clears it.
+
+        Args:
+            check_after: Time to wait after which polling the OPC status to ESR starts.
+            sleep_time: Time to sleep at each iteration in the polling loop.
+        """
+        log.debug("Clearing ESR")
```

Comment:
we should probably use logger instance that belongs to the instrument instance, right?
```suggestion
        self.log.debug("Clearing ESR")
```